File: tools/dbench/answers/python_sonnet/s1/f1_users.py
users = []
with open("users.txt") as f:
    for line in f:
        parts = line.split()
        if len(parts) == 3:
            try:
                users.append(int(parts[2]))
            except ValueError:
                pass

print(len(users))
print(sum(users) // len(users))

File: tools/dbench/answers/rust_sonnet/s3/f1_users.rs
use std::fs;

fn main() {
    let text = fs::read_to_string("users.txt").unwrap_or_default();
    let users: Vec<i64> = text.lines()
        .filter_map(|line| {
            let parts: Vec<&str> = line.split_whitespace().collect();
            if parts.len() == 3 { parts[2].parse().ok() } else { None }
        })
        .collect();
    let count = users.len();
    let avg = if count > 0 { users.iter().sum::<i64>() / count as i64 } else { 0 };
    println!("{}", count);
    println!("{}", avg);
}
